### PR TITLE
create consul version metric with version label

### PR DESCRIPTION
**Pull request by Radha13**
## Description
Create a metric  `consul_version`, e.g
```
consul_version{version="1.2.3"} = 1
```
This metric would have an unused value (always `1`) and would carry the version in its tag, so we can track it.

## Environment
```
Ubuntu 16.04, 4.4.0-142-generic
Binary built off of master
```

## Test Plan
```
radha:~$ curl -s "http://127.0.0.1:8500/v1/agent/metrics?format=prometheus" |grep -v "^#"  |grep version
consul_dub_l_radha_version{version="1.4.2"} 1
go_info{version="go1.11.4"} 1
radha:~$ 
```

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -508,6 +508,9 @@ func (a *Agent) Start() error {
 	go a.retryJoinLAN()
 	go a.retryJoinWAN()
 
+	// consul version metric with label
+	metrics.SetGaugeWithLabels([]string{"version"}, 1, []metrics.Label{{Name: "version", Value: a.config.Version}})
```
Comment (Aestek) [inline agent/agent.go]:
```suggestion
	metrics.SetGaugeWithLabels([]string{"version"}, 1, []metrics.Label{
		{Name: "version", Value: a.config.Version},
		{Name: "pre-release", Value: a.config.VersionPrerelease},
	})
```

**Comment by Radha13** [comment]
@Aestek Seems like a great idea, updated :)

**Comment by pearkes** [comment]
Thanks for the PR! Just wanted to note this would close https://github.com/hashicorp/consul/issues/4659.

**Comment by Radha13** [comment]
@pearkes Can I please get an ETA on when this PR would be merged? :bowing_woman:

**Comment by c4milo** [comment]
@pearkes, @armon, this will be very convenient to have. Is there any major concerns to get this merged?

**Comment by Radha13** [comment]
We tried this patch in our internal fork and the `-` seems to be breaking Prometheus but is processed Ok by Armon go library, hence fixing it.

**Comment by i0rek** [comment]
@Radha13 Could you rebase master so that it runs our latest test suite?

**Comment by Radha13** [comment]
Apologies for the late reply, didn't get any notification. Updated :crossed_fingers: 
Thank you for the review @i0rek

**Comment by kaxcode** [comment]
Rebased commits by accident. Github permissions would not allow undoing this. Closing this PR and moving commits changes to #9350.

**Pull request closed without merging**